src/main/java/org/aksw/word2vecrestful/word2vec/Word2VecModel.java

```diff
@@ -11,9 +17,79 @@
 
   public Map<String, float[]> word2vec;
   public int vectorSize;
-
+  public TreeSet<VectorDimension> sortedVecDimns;
+  public int[] dimRngIndxDesc;
   public Word2VecModel(final Map<String, float[]> word2vec, final int vectorSize) {
     this.word2vec = word2vec;
     this.vectorSize = vectorSize;
+    this.sortedVecDimns = new TreeSet<>();
+    //fetch vector dimension details
+    fetchVectorDimensions();
+  }
+  /**
+   * Method to fetch and store range of each dimension in the model
+   */
+  private void fetchVectorDimensions() {
+	  float[] dimVals;
+	  VectorDimension vectorDimension;
+	  //for each dimension
+	  for(int i=0;i<vectorSize;i++) {
+		  dimVals = new float[word2vec.size()];
+		  //fetch all the values
+		  int j=0;
+		  for(float[] entryVec : word2vec.values()) {
+			  dimVals[j++] =  entryVec[i];
+		  }
+		  //init and set dimension det
+		  vectorDimension = new VectorDimension(i, Word2VecMath.getMin(dimVals), Word2VecMath.getMax(dimVals));
+		  this.sortedVecDimns.add(vectorDimension);
+	  }
+	  this.dimRngIndxDesc = new int[vectorSize];
+	  Iterator<VectorDimension> descIt = sortedVecDimns.descendingIterator();
+	  int k = 0;
+	  while(descIt.hasNext()) {
+		  this.dimRngIndxDesc[k++] = descIt.next().getId();
+	  }
+  }
+  /**
+   * Method to fetch a single closest word entry to the passed input vector
+   * @param inpvec - input vector to find closest word for
+   * @return Map of the single closest word and its vector
+   */
+  public Map<String, float[]> getClosestEntry(float[] inpvec){
+	  Map<String, float[]> resMap = new HashMap<>();
+	  double minDist = -2;
+	  String minWord = null;
+	  float[] minVec = null;
+	  double tempDist;
+	  for(Entry<String, float[]> wordEntry : word2vec.entrySet()) {
+		  String word = wordEntry.getKey();
+		  float[] wordvec = wordEntry.getValue();
+		  tempDist = getEucDistSqr(inpvec, wordvec, minDist);
+		  if(tempDist != -1) {
+			  minWord = word;
+			  minVec = wordvec;
+			  minDist = tempDist;
+		  }
+	  }
+	  resMap.put(minWord, minVec);
+	  return resMap;
+  }
+  /**
+   * Method to find the squared value of euclidean distance between two vectors if it is less
+   * than the provided minimum distance value, otherwise return -1
+   * @param arr1 - first vector
+   * @param arr2 - second vector
+   * @param minDist - minimum distance constraint
+   * @return squared euclidean distance between two vector or -1
+   */
+  private double getEucDistSqr(float[] arr1, float[] arr2, double minDist) {
```

Comment:
shouldn't it be "squared Euclidian distance" instead of "Euclidian distance squared" ? So something like `getSqEucDist`? :thinking:

Reply:
Yes, that sounds better :) I will change it